src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1649,11 +1733,97 @@ var TorrentFilesTable = new Class({
             if (isPriorityComboExists(id))
                 updatePriorityCombo(id, value);
             else
-                td.adopt(createPriorityCombo(id, value));
+                td.adopt(createPriorityCombo(id, row.full_data.fileId, value));
         };
 
         this.columns['remaining'].updateTd = displaySize;
         this.columns['availability'].updateTd = displayPercentage;
+    },
+
+    getFilteredAndSortedRows: function() {
+        var sortNodes = function(nodes) {
+            nodes.sort(function(row1, row2) {
+                var column = this.columns[this.sortedColumn];
+                var res = column.compareRows(row1, row2);
+                if (this.reverseSort == '0')
+                    return res;
+                else
+                    return -res;
+            }.bind(this));
+
+            nodes.each(function(node) {
+                if (node.children.length)
+                    sortNodes(node.children);
+            });
+        }.bind(this);
+
+        var getSortedRows = function(node) {
+            sortedNodes.push(node);
+
+            node.children.each(function(child) {
+                getSortedRows(child);
+            });
+        }.bind(this);
+
+        var isFiltered = function(name) {
+            for (var i = 0; i < filterTerms.length; ++i) {
```

Comment:
how about?
```js
function(name) {
    return filterTerms.every(function(term) {
        return (name.indexOf(term) !== -1);
    });
};
```
Besides that, is it possible to just use a bool to store the filtered state?

Reply:
I've pushed new changes that "cache" the filtered set.